pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -61,8 +62,12 @@ func (e *ConditionNotFoundError) Error() string {
 // ResolvedPipelineRunTask contains a Task and its associated TaskRun, if it
 // exists. TaskRun can be nil to represent there being no TaskRun.
 type ResolvedPipelineRunTask struct {
```

Comment:
No I hadn't considered doing that. This is the direction Jason went and I didn't look into redoing it.

Reply:
OK, maybe something for future PRs. We might run loop tasks in future or pipelines as tasks, so it would be good to have a model that can work with an interface here.